shared/src/test/diff/mlscript/Methods.mls

```diff
@@ -10,20 +12,20 @@ class Bar: Foo[int, bool]
 
 Foo.Fun
 Bar.Fun
-//│ res: (foo & {Foo#B: 'B, Foo#A: 'A, y: 'B, x: 'A}) -> ('A -> 'B -> 'C) -> 'A -> 'B -> 'C
+//│ res: (foo & {Foo#B: 'B, Foo#A: 'A, y: 'B, x: 'A}) -> (nothing -> anything & 'a) -> 'a
 //│ res: (bar & {Foo#B: bool, Foo#A: int, y: bool, x: int}) -> 'a -> 'a
 
 f = Foo { x = 1; y = 2 }
 //│ f: foo & {Foo#B: 'B | 2 .. 'B, Foo#A: 'A | 1 .. 'A, y: 2, x: 1}
 
 f.(Foo.Fun)
-//│ res: (('A | 1) -> ('B | 2) -> 'C) -> 'A -> 'B -> 'C
+//│ res: (nothing -> anything & 'a) -> 'a
```

Comment:
These diffs are very fishy. I wonder what's happening here.

Reply:
I reverted the precedence fix to test the other fixes independently, then applied it back manually before committing. Turns out I applied it in the wrong place 😅